src/libraries/System.Private.Xml/src/System/Xml/Serialization/ReflectionXmlSerializationReader.cs

```diff
@@ -42,6 +42,7 @@ public ReflectionXmlSerializationReader(XmlMapping mapping, XmlReader xmlReader,
         }
 
         [RequiresUnreferencedCode(XmlSerializer.TrimSerializationWarning)]
+        [RequiresDynamicCode(XmlSerializer.AotSerializationWarning)]
```

Comment:
Would it be easier if we just moved the RUC and RDC attributes to the whole class?

Reply:
We might be able to remove the suppressions at the top of this file on line 16 and line 33.

Reply:
I think we would do the same thing to `ReflectionXmlSerializationWriter` - annotate the whole class as RDC, RUC. Then you shouldn't get any more warnings on `StringTypeDesc` and `QnameTypeDesc`

Reply:
Totally my opinion from looking at the code quick (I haven't done much development in this area). It is just easier to annotate and reason about when you slap the attributes on the whole class when the class's whole purpose is to do trim/AOT-incompatible stuff. In both these cases ("ReflectionReader/Writer" and "CodeGenerator") their whole purpose is to use reflection and IL Ref Emit.